### ISSUE TITLE
Stuck getting cluster list

### ISSUE THREAD

**Issue by cehoffman** [coreos/etcd-operator#1338]
ETCD operator 0.4.2 becomes unable to perform cluster maintenance operations like scaling due to following error log.

> time="2017-08-01T02:26:31Z" level=error msg="initialization failed: json: cannot unmarshal number 1.003039e+06 into Go struct field ClusterListCopy.items of type int64" pkg=controller 

The cluster resource is:

```
apiVersion: etcd.coreos.com/v1beta1
kind: Cluster
metadata:
  name: portworx-etcd
  namespace: portworx
  selfLink: /apis/etcd.coreos.com/v1beta1/namespaces/portworx/clusters/portworx-etcd
  uid: 6e5e0f36-758e-11e7-ad3d-000d3a03c20a
  resourceVersion: '134756084'
  creationTimestamp: '2017-07-31T01:20:25Z'
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: >
      {"apiVersion":"etcd.coreos.com/v1beta1","kind":"Cluster","metadata":{"annotations":{},"name":"portworx-etcd","namespace":"portworx"},"spec":{"TLS":{"static":{"member":{"peerSecret":"portworx-etcd-peer-tls","serverSecret":"portworx-etcd-server-tls"},"operatorSecret":"portworx-etcd-client-tls"}},"backup":{"backupIntervalInSecond":1800,"maxBackups":5,"s3":{"awsSecret":"portworx-backup","s3Bucket":"omni-portworx-backup"},"storageType":"S3"},"pod":{"antiAffinity":true,"etcdEnv":null,"nodeSelector":{"node-role.kubernetes.io/master":""},"resources":{},"tolerations":[{"effect":"NoSchedule","key":"node-role.kubernetes.io/master","operator":"Exists"}]},"restore":{"backupClusterName":"portworx-etcd","s3":{"awsSecret":"portworx-backup","s3Bucket":"omni-portworx-backup"},"storageType":"S3"},"size":3,"version":"3.1.10"}}
spec:
  TLS:
    static:
      member:
        peerSecret: portworx-etcd-peer-tls
        serverSecret: portworx-etcd-server-tls
      operatorSecret: portworx-etcd-client-tls
  backup:
    autoDelete: false
    backupIntervalInSecond: 1800
    maxBackups: 5
    s3:
      awsSecret: portworx-backup
      s3Bucket: omni-portworx-backup
    storageType: S3
  baseImage: quay.io/coreos/etcd
  pod:
    antiAffinity: true
    nodeSelector:
      node-role.kubernetes.io/master: ''
    resources: {}
    tolerations:
      - effect: NoSchedule
        key: node-role.kubernetes.io/master
        operator: Exists
  restore:
    backupClusterName: portworx-etcd
    storageType: S3
  size: 3
  version: 3.1.10
status:
  size: 3
  reason: ''
  backupServiceStatus:
    backupSize: 179.559
    backups: 6
    recentBackup:
      creationTime: '2017-07-31T13:48:44Z'
      revision: 1003039
      size: 123.359
      timeTookInSecond: 8
      version: 3.1.10
  targetVersion: ''
  controlPaused: false
  conditions:
    - reason: ''
      transitionTime: '2017-07-31T01:20:33Z'
      type: Recovering
    - reason: 'Current cluster size: 1, desired cluster size: 3'
      transitionTime: '2017-07-31T01:22:18Z'
      type: ScalingUp
    - reason: 'Current cluster size: 2, desired cluster size: 3'
      transitionTime: '2017-07-31T01:22:31Z'
      type: ScalingUp
    - reason: 'Current cluster size: 2, desired cluster size: 3'
      transitionTime: '2017-07-31T01:22:39Z'
      type: ScalingUp
    - reason: ''
      transitionTime: '2017-07-31T01:22:53Z'
      type: Ready
  members:
    ready:
      - portworx-etcd-0001
      - portworx-etcd-0000
      - portworx-etcd-0002
  phase: Running
  currentVersion: 3.1.10
```

I believe it is complaining about `.status.backupServiceStatus.recentBackup.revision`.

**Comment by hongchaodeng** [comment]
@cehoffman 
Thanks for trying out and reporting the problem. Can you tell us how to reproduce the bug?

Also, mind try to use v0.5.0 instead? It replaces TPR with CRD.

> ClusterListCopy

This thing is some workaround for TPR bug and TPR is deprecated in k8s 1.7+ .

**Comment by cehoffman** [comment]
It'll be a few days to a week before I can try v0.5.0. Waiting on the next tectonic release to get 1.7.

I got to the bug from as best as I can tell enabling s3 backup and restore. I used the restore twice I think over about 5 days. I also had to manually run compaction on the etcd cluster when it reached an alarm stay from keyspace size. I did this a few more times. I notice this state after I was trying to migrate the cluster purposefully from a few nodes that were being removed.
